dicthash/dicthash.py

```diff
@@ -51,8 +55,8 @@ def _generate_string_from_iterable(l):
     """
     # we need to handle strings separately to avoid infinite recursion
     # due to their iterable property
-    if isinstance(l, (str, unicode)):
-        return unicode(l)
+    if isinstance(l, basestring):
```

Comment:
See above. We are actually checking for `str`. The code snippet above ensure that we are always checking for the right `str`.